### ISSUE TITLE
SWaT   processing 

### ISSUE THREAD

**Issue by WSake** [17000cyh/IMDiffusion#7]
各位大佬，请问一下有没有SWaT数据集的处理代码~   用其他论文的代码，最后结果很低！

**Comment by SinChee** [comment]
The paper said that SWaT dataset is ran with 51 channels, but in the code, it is ran with 45 channels (refer to line 123 in exe_machine.py). I attached a snippet from their paper as follows.

<img width="659" alt="Image" src="https://github.com/user-attachments/assets/0dfebace-2735-4f62-a04b-4493bee609e7" />

My team notices this and change the code in line 123 to 51 and ran the code again and attempt to replicate their results. However, we encounter gradient explosion at the first epoch training. The input is normal, but the weights of the models are `nan`. We tried gradient clipping in the train loop, the convergence become very slow and eventually we are faced with gradient explosion again. 

Can the author of this paper enlighten us the differences in code implementation and your paper?

**Comment by 17000cyh** [comment]
Hi, the original SWaT dataset contains 51 channels. However, we noticed that 6 of these channels have completely constant values (equal to 1) in both the training and test sets. To optimize computational efficiency, we removed these channels.

2025-03-26 14:25:59 "Sin Chee" ***@***.***> 写道：

The paper said that SWaT dataset is ran with 51 channels, but in the code, it is ran with 45 channels (refer to line 123 in exe_machine.py). I attached a snippet from their paper as follows.

Screenshot.2025-03-26.at.14.19.12.png (view on web)

My team notices this and change the code in line 123 to 51 and ran the code again and attempt to replicate their results. However, we encounter gradient explosion at the first epoch training. The input is normal, but the weights of the models are nan. We tried gradient clipping in the train loop, the convergence become very slow and eventually we are faced with gradient explosion again.

Can the author of this paper enlighten us the differences in code implementation and your paper?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you are subscribed to this thread.Message ID: ***@***.***>

SinChee left a comment (17000cyh/IMDiffusion#7)

The paper said that SWaT dataset is ran with 51 channels, but in the code, it is ran with 45 channels (refer to line 123 in exe_machine.py). I attached a snippet from their paper as follows.

Screenshot.2025-03-26.at.14.19.12.png (view on web)

My team notices this and change the code in line 123 to 51 and ran the code again and attempt to replicate their results. However, we encounter gradient explosion at the first epoch training. The input is normal, but the weights of the models are nan. We tried gradient clipping in the train loop, the convergence become very slow and eventually we are faced with gradient explosion again.

Can the author of this paper enlighten us the differences in code implementation and your paper?

**Comment by SinChee** [comment]
Thank you for replying, does this mean that the rest of the comparison in your paper is also processed this way?
